pallets/contracts/src/lib.rs

```diff
@@ -333,7 +338,7 @@ decl_module! {
         /// - All the errors in `pallet_contracts::Call::instantiate_with_code` can also happen here.
         /// - CDD/Permissions are checked, unlike in `pallet_contracts`.
         /// - Errors that arise when adding a new secondary key can also occur here.
-        #[weight = Module::<T>::weight_instantiate_with_code(&code, &salt, &perms).saturating_add(*gas_limit)]
+        #[weight = Module::<T>::weight_instantiate_with_code(&code, &salt, &Some(perms.clone())).saturating_add(*gas_limit)]
```

Comment:
```suggestion
        #[weight = Module::<T>::weight_instantiate_with_code(&code, &salt, Some(&perms)).saturating_add(*gas_limit)]
```
Don't need a `clone` here.  Option can wrap a reference.